cmd/expiration-mailer/main.go

```diff
@@ -54,35 +56,57 @@ type mailer struct {
 	clk           clock.Clock
 }
 
-func (m *mailer) sendNags(parsedCert *x509.Certificate, contacts []*core.AcmeURL) error {
-	expiresIn := int(parsedCert.NotAfter.Sub(m.clk.Now()).Hours() / 24)
+func (m *mailer) sendNags(contacts []*core.AcmeURL, certs []*x509.Certificate) error {
+	if len(contacts) == 0 {
+		return nil
+	}
+	if len(certs) == 0 {
+		return errors.New("no certs given to send nags for")
+	}
 	emails := []string{}
 	for _, contact := range contacts {
 		if contact.Scheme == "mailto" {
 			emails = append(emails, contact.Opaque)
 		}
 	}
-	if len(emails) > 0 {
-		email := emailContent{
-			ExpirationDate:   parsedCert.NotAfter,
-			DaysToExpiration: expiresIn,
-			DNSNames:         strings.Join(parsedCert.DNSNames, ", "),
-		}
-		msgBuf := new(bytes.Buffer)
-		err := m.emailTemplate.Execute(msgBuf, email)
-		if err != nil {
-			m.stats.Inc("Mailer.Expiration.Errors.SendingNag.TemplateFailure", 1, 1.0)
-			return err
-		}
-		startSending := m.clk.Now()
-		err = m.mailer.SendMail(emails, m.subject, msgBuf.String())
-		if err != nil {
-			m.stats.Inc("Mailer.Expiration.Errors.SendingNag.SendFailure", 1, 1.0)
-			return err
+	if len(emails) == 0 {
+		return nil
+	}
+
+	expiresIn := time.Duration(math.MaxInt64)
+	expDate := m.clk.Now()
+	domains := []string{}
+
+	for _, cert := range certs {
```

Comment:
This loop could use a summary comment about what it's doing. For instance, it's not immediately clear what the variable name `nextOne` means.